workflow/controller/operator.go

```diff
@@ -2108,6 +2108,23 @@ func (woc *wfOperationCtx) markNodePhase(nodeName string, phase wfv1.NodePhase,
 	return node
 }
 
+func (woc *wfOperationCtx) getPodByNode(node *wfv1.NodeStatus) (*apiv1.Pod, error) {
+	if node.Type != wfv1.NodeTypePod {
+		return nil, fmt.Errorf("Expected node type %s, got %s", wfv1.NodeTypePod, node.Type)
+	}
+	podList, err := woc.getAllWorkflowPods()
```

Comment:
Tthere is a faster way to do this. Use the `woc.controller.podInformer`. I think it is `woc.controller.podInformer.GetStore().GetKeyKey(woc.wf.Namepsace+"/"+node.ID)`